src/sagemaker/lineage/artifact.py

```diff
@@ -522,3 +584,75 @@ def endpoint_contexts(
         for vertex in query_result.vertices:
             endpoint_contexts.append(vertex.to_lineage_object())
         return endpoint_contexts
+
+    def upstream_datasets(
+        self, direction: LineageQueryDirectionEnum = LineageQueryDirectionEnum.ASCENDANTS
```

Comment:
Same comment as above re: removing direction and moving common code to internal function